src/main/java/io/stargate/sgv2/jsonapi/service/bridge/executor/CollectionSettings.java

```diff
@@ -53,6 +59,52 @@ public static SimilarityFunction fromString(String similarityFunction) {
     }
   }
 
+  public static CollectionSettings getCollectionSettings(
+      TableMetadata table, ObjectMapper objectMapper) {
+    String collectionName = table.getName().asCql(true);
+    // get vector column
+    final Optional<ColumnMetadata> vectorColumn =
+        table.getColumn(DocumentConstants.Fields.VECTOR_SEARCH_INDEX_COLUMN_NAME);
+    boolean vectorEnabled = vectorColumn.isPresent();
+    // if vector column exist
+    if (vectorEnabled) {
+      final int vectorSize = ((VectorType) vectorColumn.get().getType()).getDimensions();
+      // get vector index
+      IndexMetadata vectorIndex = null;
+      Map<CqlIdentifier, IndexMetadata> indexMap = table.getIndexes();
+      for (CqlIdentifier key : indexMap.keySet()) {
+        if (key.asInternal().contains(DocumentConstants.Fields.VECTOR_SEARCH_INDEX_COLUMN_NAME)) {
+          vectorIndex = indexMap.get(key);
+          break;
+        }
+      }
+      // default function
+      CollectionSettings.SimilarityFunction function = CollectionSettings.SimilarityFunction.COSINE;
+      if (vectorIndex != null) {
+        final String functionName =
+            vectorIndex.getOptions().get(DocumentConstants.Fields.VECTOR_INDEX_FUNCTION_NAME);
+        if (functionName != null)
+          function = CollectionSettings.SimilarityFunction.fromString(functionName);
+      }
+      final String comment = (String) table.getOptions().get("comment");
+      if (comment != null && !comment.isBlank()) {
+        return createCollectionSettingsFromJson(
+            collectionName, vectorEnabled, vectorSize, function, comment, objectMapper);
+      } else {
+        return new CollectionSettings(
+            collectionName, vectorEnabled, vectorSize, function, null, null);
+      }
+    } else { // if not vector collection
+      return new CollectionSettings(
+          collectionName,
+          vectorEnabled,
+          0,
+          CollectionSettings.SimilarityFunction.UNDEFINED,
+          null,
+          null);
+    }
+  }
+
   public static CollectionSettings getCollectionSettings(
```

Comment:
Can we get this method out and change the create collection use the above one in the same PR?